content/ember-v4-typescript-support-update.md

```diff
@@ -0,0 +1,108 @@
+
+    * Upgrade to `ember-qunit@^5` and `@types/ember-qunit@^5`.
+
+    * Anywhere you have imported `TestContext` from `ember-test-helpers`, replace it with `import { type TestContext } from '@ember/test-helpers';`. You will likely be able to merge this with existing imports from `@ember/test-helpers`.
+
+    * Remove all references to `ember-test-helpers` from your project, and remove `@types/ember-test-helpers` from your dependencies.
+
+Once everything is compiling again, you will be good to go!
+
+
+## What if I get stuck?
+
+You may hit interesting challenges in dealing with the improved type strictness in places where your code relied on the loose types (or the erroneous types!) before. If you are having a hard time, please reach out in [#e-typescript on Discord][e-typescript] or [start a Discussion on the ember-cli-typescript-repo](https://github.com/typed-ember/ember-cli-typescript/discussions). The Typed Ember team is happy to help, and many other people in the community are as well.
+
+Also, while we’ve done our very best to make this totally bug-free , it’s entirely possible that we missed something or made a mistake! Please feel free to check in with us [on Discord][e-typescript] or to [file a bug](https://github.com/typed-ember/ember-cli-typescript/issues). We encourage you to check with us first, though: in many cases things which seem like bugs are actually the type system catching errors the previous types couldn’t catch!
+
+
+## The future
+
+We hope (and fully expect!) this to be the last major release of Ember’s types on DefinitelyTyped. During the 4.x series, we aim to finish the work to [officially support TypeScript](https://github.com/emberjs/rfcs/pull/724). That will make it much easier for TypeScript users (and maintainers!) to deal with both major and minor updates:
+
+* Deprecations will automatically be present in both types and run-time code, and they will be removed right alongside runtime changes when prepping for a major release.
+
+* The design of deprecations or changes will include smooth migration paths for types as well as for the runtime behavior of the app.
+
+* More generally, having types as first-class parts of every design discussion means that new APIs will be TypeScript-friendly automatically. (Many older Ember APIs are *very* unfriendly to TypeScript!)
+
+* <!-- alex ignore just --> The supported TypeScript versions will be documented clearly, just like with Node and browsers, making it easier for you to know what you should do to stay up to date.
+
+* Breaking changes in one package will not necessarily cause breaking changes in other packages. The updates to the test infrastructure types, for example, will not breaking in the future!
+
+And as always, those benefits will extend to *even more* of our just-JavaScript users, because many more editors take advantage of types which ship natively with packages than implicitly take advantage of the types from DefinitelyTyped.
```

Comment:
native JavaScript? vanilla JavaScript? platform JavaScript :trollface:?


```suggestion
And as always, those benefits will extend to *even more* of our JavaScript users, because many more editors take advantage of types which ship natively with packages than implicitly take advantage of the types from DefinitelyTyped.
```